packages/react/src/SelectPanel/SelectPanel.tsx

```diff
@@ -647,6 +710,13 @@ function Panel({
             '--max-height': overlayProps?.maxHeight ? heightMap[overlayProps.maxHeight] : heightMap['large'],
             /* override AnchoredOverlay position */
             transform: variant === 'modal' ? 'translate(-50%, -50%)' : undefined,
+            // set maxHeight based on calculated availablePanelHeight when keyboard is visible
+            ...(keyboardState.isVisible
+              ? {
+                  maxHeight:
+                    typeof availablePanelHeight === 'number' ? `${availablePanelHeight}px` : availablePanelHeight,
+                }
+              : {}),
```

Comment:
```suggestion
            // set maxHeight based on calculated availablePanelHeight when keyboard is visible
             maxHeight: `${availablePanelHeight}px` ?? 'auto'
```


if you set the `availablePanelHeight` type to `number | undefined` , can also eliminate else statement in line 439 in favor of `setAvailablePanelHeight(window.visualViewport?.height)`